### ISSUE TITLE
v2.8.4 seems to remove all query string parameters

### ISSUE THREAD

**Issue by tamsler** [restify/node-restify#739]
I just upgraded from v2.8.3 to v2.8.4, and the new version (v2.8.4) seems to remove all query string parameters. Not quite sure what's going on. All my tests are failing indicating that the query string parameters are not present.

**Comment by gergelyke** [comment]
wow, that's strange - what os and what node version are you using?

**Comment by tamsler** [comment]
It is strange. I am running on Mac OS X 10.9 and 10.10
Node: v0.10.36

**Comment by DonutEspresso** [comment]
Much has changed since this issue originally filed, @tamsler can you confirm if this is still a problem? If so, can you please provide a gist that reproduces the problem?

**Comment by tamsler** [comment]
I have some more information on this. So the doc indicates that getQuery() returns the raw query string. If you want an object containing the data, use the queryParser. The queryParser object indicates that the parsed content will always be available in req.query.  In the 1.4 to 2.0 migration document, it says that:
https://github.com/restify/node-restify/wiki/1.4-to-2.0-Migration-Tips#reqquery---reqgetquery-or-reqquery
req.query -> req.getQuery() or req.query()

So I used req.getQuery() everywhere I used req.query before. Up until v2.8.4, req.getQuery() returned a parsed queryString object. Starting with v2.8.5, it just returns a queryString string.
